src/azul/strings.py

```diff
@@ -147,3 +152,9 @@ def trunc_ellipses(s: STRING, /, max_len: int) -> STRING:
         s = s[:max_len - len(ellipses)] + ellipses
     assert len(s) <= max_len, (len(s), max_len)
     return s
+
+
+def reversible_join(joiner: str, parts: Iterable[str]):
```

Comment:
Let's not make this a library function. If we did, I'd have a few more concerns that I don't want to spend time on now.